### PR TITLE
Remove imagemin & add a limit on the number of tasks run simultaneously

### PR DESCRIPTION
**Pull request by 343dev**

### CODE REVIEW

**Review by igoradamenko — CHANGES_REQUESTED** [review]

File: lib/createWebp.js [hunk 1]
```diff
@@ -1,67 +1,81 @@
...
+  const totalPaths = paths.length;
+
+  if (!totalPaths) return;
+
+  spinner.start(`Creating WebP for ${totalPaths} ${getPlural(totalPaths, 'image', 'images')}...`);
+
+  const limit = pLimit(os.cpus().length);
+
+  const tasks = paths.map(filePath => {
+    const readFile = promisify(fs.readFile);
```
[the remark this answers is not in the document]
Comment (343dev) [reply lib/createWebp.js:25]: Replaced `promisify(fs.readFile)` with `fs.promises.readFile`. Thanks.

File: lib/optimize.js [hunk 2]
```diff
@@ -1,156 +1,186 @@
...
+
   spinner.clear();
+  tasksResult.forEach(({ fileBuffer, filePath }) => checkResult(fileBuffer, filePath));
+
+  console.log();
+  spinner.succeed('Optimizing completed');
+}
+
+function checkResult(fileBuffer, filePath) {
+  if (!Buffer.isBuffer(fileBuffer) || typeof filePath !== 'string') return;
```
[the remark this answers is not in the document]
Comment (343dev) [reply lib/optimize.js:58]: This is no fine. Added. Thanks.

File: lib/createWebp.js [hunk 3]
```diff
@@ -1,67 +1,81 @@
...
+      .then(fileBuffer => sharp(fileBuffer)
+        .webp({
+          quality: 85,
+          lossless,
+        })
+        .toBuffer())
+      .then(fileBuffer => ({ fileBuffer, filePath })));
   });
 
+  const tasksResult = await Promise.all(tasks);
```
[the remark this answers is not in the document]
Comment (343dev) [reply lib/createWebp.js:37]: Seems like `Promise.all` works fine here 🤔

File: lib/optimize.js [hunk 4]
```diff
@@ -1,156 +1,186 @@
...
-    At the time of lossless optimization Guetzli encoder will be used,
-    and it converts PNGs to JPEGs.
-   */
-  const jpeg = await imagemin(pathsByExt.jpeg, {
-    plugins: lossless
-      ? [imageminGuetzli({ quality: 90 })]
-      : [imageminJpegoptim({ progressive: true, max: 80 })],
-  });
+  const limit = pLimit(
+    isLossless ? Math.round(os.cpus().length / 2) : os.cpus().length,
```
[the remark this answers is not in the document]
Comment (343dev) [reply lib/optimize.js:33]: Added.

File: lib/createWebp.js [hunk 5]
```diff
@@ -1,67 +1,81 @@
...
+  if (!totalPaths) return;
+
+  spinner.start(`Creating WebP for ${totalPaths} ${getPlural(totalPaths, 'image', 'images')}...`);
+
+  const limit = pLimit(os.cpus().length);
+
+  const tasks = paths.map(filePath => {
+    const readFile = promisify(fs.readFile);
+
+    return limit(() => readFile(filePath)
```
[the remark this answers is not in the document]
Comment (343dev) [reply lib/createWebp.js:27]: Added.

**Comment by igoradamenko** [comment]
Published @funboxteam/optimizt@2.3.0 🚀

**Pull request merged**